bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -102,14 +195,29 @@ public BigtableBufferedMutator(
         options.getChannelCount(),
         heapSizeManager.getMaxHeapSize(),
         heapSizeManager.getMaxInFlightRpcs());
+    this.options = options;
+    this.heapSizeManager = heapSizeManager;
+    this.executorService = executorService;
+    initializeAsyncMutators();
+  }
+
+  private void initializeAsyncMutators() {
+    for (int i = activeMutationWorkers.get(); i < options.getAsyncMutatorCount(); i++) {
+      executorService.submit(mutationWorker);
+    }
   }
 
   @Override
   public void close() throws IOException {
+    flush();
```

Comment:
There's a subtle set of interactions here.  There's an expectation that if heapSizeManager.flush() is complete, mutationsToBeSent queue will be empty.  I can walk you through the details in person.